src/cli/cli_config.h

```diff
@@ -99,4 +99,14 @@
 #error "Unsupported CLI transport!"
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MAX_USER_COMMAND_NUMBER
+ *
+ * The maxium number of supported user commands.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_MAX_USER_COMMAND_NUMBER
+#define OPENTHREAD_CONFIG_MAX_USER_COMMAND_NUMBER 10
```

Comment:
Suggest using a smaller default number here in the config to avoid using memory? I would say 2 maybe?